### PR TITLE
Updating PR template and Contributing document

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
## Description

Additional information for submitting PRs was requested: https://github.com/AgPipeline/.github/pull/6. This is the implementation of those requests

Implementation for PRs:
* https://github.com/AgPipeline/issues-and-projects/issues/186
* https://github.com/AgPipeline/issues-and-projects/issues/182

## Types of changes

The Contributing document had legacy text which was changed/updated/removed
Added instructions regarding filling out a PR, especially around CI (Continuous Integration).
Added links to OSF in Contributing doc
Added link to Contributing doc, PR section, to PR template
Updated OSF page on Pull Requests: https://osf.io/tzmhp/wiki/Issues%20and%20Pull%20Requests/

## Checklist:

- [x] I have read the documentation on [CONTRIBUTING](https://github.com/AgPipeline/Organization-info) 
- [ ] My updates require a change to the documentation 
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## Time frame for completing Pull Request review (from creation date):
- [ ] As soon as possible
- [x] In the next two weeks
- [ ] In the next month

## Focus of Pull Request:

Reviewers should focus on the following aspects of this pull request. All comments are welcome regardless of which items are checked off.
- [ ] Inform others of new features and changes
- [ ] There are no errors or inconsistencies
- [x] Everything appears clear and easy to understand
- [x] Suggest improvements
- [x] Identify areas where functionality and/or documentation is missing

## Hints to reviewers:

Most of the changes are regarding information to put into PRs.
@dlebauer in addition to reviewing the document, determine if this includes the information you expected
@KristinaRiemer, @MagicMilly , @julianpistorius Review for consistency and readability as well

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -10,15 +10,15 @@ There are many ways to get in touch:
...
-  * reference data: github.com/terraref/reference-data/issues
+* join discussion of open issues, features, and bugs at AgPipeline/issues-and-projects/issues
 * contribute algorithms to the pipeline
 * revise documentation
 
-New features are directed toward developing and extending the TERRA Ref computational infrastructure. 
+New features are directed toward developing and extending the AgPipeline computational infrastructure. 
 
-If you need any help, please contact us in our [chat room](https://gitter.im/terraref/reference-data) or creating a [new issue](https://github.com/terraref/computing-pipeline/issues/new).
+If you need any help, please create a [new issue](https://github.com/terraref/computing-pipeline/issues/new).
```
Comment (dlebauer) [inline .github/CONTRIBUTING.md]: maybe give other options like emailing the maintainer? searching issues-and-templates issues to see if it has come up before?

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -29,34 +29,27 @@ In these cases, please link the issues across repositories (e.g. add `will be im
...
-        - Create an issue in GitHub, assuming one does not already exist.
+    - Create an issue in GitHub, assuming one does not already exist.
 	- Clearly describe the issue including steps to reproduce when it is a bug.
 	- Make sure you fill in the earliest version that you know has the issue.
-- Ask @dlebauer or @robkooper to add you to the TERRA Ref project if you plan on fixing the issue.
-
-* Github Issues
-  * [computing pipeline (infrastructure, algorithms)](github.com/terraref/computing-pipeline/issues/new)
-  * [reference data (data products)](github.com/terraref/reference-data/issues/new)
+- Ask @dlebauer or @chris-schnaufer to add you to the AgPipeline project if you plan on fixing the issue.
```
Comment (dlebauer) [inline .github/CONTRIBUTING.md]: is this necessary within a fork + PR workflow? I thought that one of the advantages of using forks is that people don't need to be added to the organization ...

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -29,34 +29,27 @@ In these cases, please link the issues across repositories (e.g. add `will be im
...
 
 ### Overview
 
 When you add a significant **new feature**, please create an issue first, to allow others to comment and give feedback. 
 
-When you have created a new feature or non-trivial change to existing code, create a 'pull request'.
+When you have created a new feature or non-trivial change to existing code, create a [Pull Request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).
 
-**Branching and Pull Requests**: Although many core developers have write permissions on the TERRA Ref repositories, 
+**Branching and Pull Requests**: Although many core developers have write permissions on the AgPipeline repositories,
```
Comment (dlebauer) [inline .github/CONTRIBUTING.md]: Should we encouraging forking?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -67,95 +60,94 @@ If you haven't used git before, the GitHub website and GitHub desktop client all
 Any file can be edited in the GitHub interface, and new files can be created. 
 GitHub will create these as a new pull request.
 
-### Using Git at the Command Line
+### Using Git at the Command Line for GitHub
```
Comment (dlebauer) [inline .github/CONTRIBUTING.md:68]: * a lot of this is specific to the issues-and-projects repository even though the PR template is for the entire system
* At this point there are so many tutorials on using git ... would it be sufficient to link to documentation maintained by github, e.g. https://guides.github.com/activities/forking/ or this course https://lab.github.com/githubtraining/introduction-to-github? or, should this perhaps go in the OSF page about Github?

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md]: Good suggestion. Adding

Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md:19]: Changed to reflect this and also added more links to our OSF page on Issues and Pull Requests: https://osf.io/tzmhp/wiki/Issues%20and%20Pull%20Requests/

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md:42]: Changed to reflect this and added more links to our OSF page on Issues and Pull Requests: https://osf.io/tzmhp/wiki/Issues%20and%20Pull%20Requests/

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md]: Changed to reflect this and added more links to our OSF page on Issues and Pull Requests: https://osf.io/tzmhp/wiki/Issues%20and%20Pull%20Requests/

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md]: Good catch: not needed

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md:68]: This is out of scope for the issue being merged: https://app.zenhub.com/workspaces/ua-ag-data-science-5a57a3198339f11ba1c85775/issues/agpipeline/issues-and-projects/186

Updated PR to reflect which issue it's tied to.

I will create a ticket for this comment: https://app.zenhub.com/workspaces/ua-ag-data-science-5a57a3198339f11ba1c85775/issues/agpipeline/issues-and-projects/218

Comment (Chris-Schnaufer) [reply .github/CONTRIBUTING.md:68]: Also added an issue for adding a section to the PR template for linking to the issue a PR is for: https://app.zenhub.com/workspaces/ua-ag-data-science-5a57a3198339f11ba1c85775/issues/agpipeline/issues-and-projects/219

**Comment by Chris-Schnaufer** [comment]
Merging due to lack of additional comments

**Pull request merged**